iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -972,14 +972,43 @@ private boolean isOrcOnlyFiles(org.apache.hadoop.hive.metastore.api.Table hmsTab
   }
 
   // TODO: remove this if copy-on-write mode gets implemented in Hive
-  private static void setDeleteModeOnTableProperties(Table icebergTable, Map<String, String> newProps) {
+  private void setDeleteModeOnTableProperties(Table icebergTbl, Map<String, String> newProps,
+      EnvironmentContext context) {
     // Hive only supports merge-on-read delete mode, it will actually throw an error if DML operations are attempted on
     // tables that don't have this (the default is copy-on-write). We set this at table creation and v1->v2 conversion.
-    if ((icebergTable == null || ((BaseTable) icebergTable).operations().current().formatVersion() == 1) &&
+    if ((icebergTbl == null || ((BaseTable) icebergTbl).operations().current().formatVersion() == 1) &&
         IcebergTableUtil.isV2Table(newProps)) {
-      newProps.put(TableProperties.DELETE_MODE, HiveIcebergStorageHandler.MERGE_ON_READ);
-      newProps.put(TableProperties.UPDATE_MODE, HiveIcebergStorageHandler.MERGE_ON_READ);
-      newProps.put(TableProperties.MERGE_MODE, HiveIcebergStorageHandler.MERGE_ON_READ);
+      if (catalogProperties.get(TableProperties.DELETE_MODE) == null) {
+        catalogProperties.put(TableProperties.DELETE_MODE, HiveIcebergStorageHandler.MERGE_ON_READ);
+        newProps.put(TableProperties.DELETE_MODE, HiveIcebergStorageHandler.MERGE_ON_READ);
+      }
+      if (catalogProperties.get(TableProperties.UPDATE_MODE) == null) {
+        catalogProperties.put(TableProperties.UPDATE_MODE, HiveIcebergStorageHandler.MERGE_ON_READ);
+        newProps.put(TableProperties.UPDATE_MODE, HiveIcebergStorageHandler.MERGE_ON_READ);
+      }
+      if (catalogProperties.get(TableProperties.MERGE_MODE) == null) {
+        catalogProperties.put(TableProperties.MERGE_MODE, HiveIcebergStorageHandler.MERGE_ON_READ);
+        newProps.put(TableProperties.MERGE_MODE, HiveIcebergStorageHandler.MERGE_ON_READ);
+      }
+
+      if (context != null) {
+        Splitter splitter = Splitter.on(PROPERTIES_SEPARATOR);
+        Map<String, String> contextProperties = context.getProperties();
+        if (contextProperties.containsKey(SET_PROPERTIES)) {
+          String propValue = context.getProperties().get(SET_PROPERTIES);
+          List propsList = splitter.splitToList(propValue);
+          if (!propsList.contains(TableProperties.DELETE_MODE)) {
+            propValue += "'" + TableProperties.DELETE_MODE;
```

Comment:
where is the closing quote?
could this be simplified to 
````
splitter.splitToList(contextProperties.get(SET_PROPERTIES))
  .forEach(k -> {
    switch(k){
       case UPDATE_MODE:
          propValue += "'" + TableProperties.UPDATE_MODE;
          break;
...
     }
})
````

Reply:
The value of property `SET_PROPERTIE`  use single quotes to make up relevant table proerties. e.g.
`last_modified_time'format-version'last_modified_by'write.delete.mode'write.update.mode'write.merge.mode`

Reply:
thanks for the clarification